ark/api/v1alpha1/a2atask_types.go

```diff
@@ -0,0 +1,226 @@
+	// +kubebuilder:validation:MinLength=1
+	TaskID string `json:"taskId"`
+	// ContextID links this task to an A2A conversation context for stateful interactions.
+	// +kubebuilder:validation:Optional
+	ContextID string `json:"contextId,omitempty"`
+	// Input contains the user's input that initiated this task.
+	// +kubebuilder:validation:Optional
+	Input string `json:"input,omitempty"`
+	// Parameters contains additional key-value parameters for task execution.
+	// +kubebuilder:validation:Optional
+	Parameters map[string]string `json:"parameters,omitempty"`
+	// Priority determines task execution order (higher values execute first).
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=0
+	Priority int32 `json:"priority,omitempty"`
+	// Timeout specifies how long we will poll the A2A server for task completion before timing out.
+	// If the task has not reached a terminal state (completed, failed, cancelled) within this duration,
+	// it will be marked as failed.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="12h"
+	Timeout *metav1.Duration `json:"timeout,omitempty"`
+	// TTL (time to live) specifies how long to keep this A2ATask resource in the system after completion.
+	// After this duration, the resource may be automatically deleted.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="720h"
+	TTL *metav1.Duration `json:"ttl,omitempty"`
+	// PollInterval specifies how frequently to check the A2A server for task status updates.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="5s"
+	PollInterval *metav1.Duration `json:"pollInterval,omitempty"`
+}
+
+// A2ATaskStatus defines the observed state of an A2ATask.
+// Combines Kubernetes lifecycle tracking with A2A protocol task data.
+type A2ATaskStatus struct {
+	// Phase indicates the current Kubernetes lifecycle stage of the task.
+	// Possible values: pending, assigned, running, input-required, auth-required, completed, failed, cancelled, unknown.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="pending"
+	// +kubebuilder:validation:Enum=pending;assigned;running;input-required;auth-required;completed;failed;cancelled;unknown
+	Phase string `json:"phase,omitempty"`
+	// Conditions represent the latest available observations of the task's state.
+	// The Completed condition indicates whether the task is no longer running.
+	// +kubebuilder:validation:Optional
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+	// StartTime records when task execution began.
+	// +kubebuilder:validation:Optional
+	StartTime *metav1.Time `json:"startTime,omitempty"`
+	// CompletionTime records when task execution finished (success or failure).
+	// +kubebuilder:validation:Optional
+	CompletionTime *metav1.Time `json:"completionTime,omitempty"`
+	// Error contains the error message if the task failed.
+	// +kubebuilder:validation:Optional
+	Error string `json:"error,omitempty"`
```

Comment:
this error is not needed right, normally that'll be in the condition - I don't think we have error for query, pls check